scripts/flow_track_correl.py

```diff
@@ -0,0 +1,414 @@
+import numpy as np
+from astropy.io import fits
+import glob
+import matplotlib
+import matplotlib.cm as cm
+import matplotlib.pyplot as plt
+import os
+
+
+def read_fits(filepath: str, hdu: int = 1) -> np.ndarray:
+    """
+    Read FITS image array.
+
+    Parameters
+    ----------
+    filepath : str
+        Path to the FITS file.
+    hdu : int, optional
+        Index of the header data unit storing the data array, by default 1.
+
+    Returns
+    -------
+    np.ndarray
+        2D image array from the specified FITS file.
+    """
+    with fits.open(filepath) as hdul:
+        hdul.verify('fix')
+        data = hdul[hdu].data
+    return data
+
+
+def read_header(filepath: str, hdu: int = 1) -> fits.header.Header:
+    """
+    Read the header from a FITS file.
+
+    Parameters
+    ----------
+    filepath : str
+        Path to the FITS file.
+    hdu : int, optional
+        Index of the header data unit containing the header information, by default 1.
+
+    Returns
+    -------
+    fits.header.Header
+        Header object from the specified FITS file.
+    """
+    with fits.open(filepath) as hdul:
+        hdul.verify('fix')
+        header = hdul[hdu].header
+    return header
+
+
+def calc_ylims(ycen_band_rs: np.ndarray, band_width: float, header: fits.header.Header) -> [int, int]:
+    """
+    Convert y-coordinates of lower and upper row of bands to array indices for slicing.
+
+    Parameters
+    ----------
+    ycen_band_rs : np.ndarray
+        y-coordinates of center of band in solar radii.
+
+    band_width : float
+        Half-width of the band in solar radii.
+
+    header : fits.header.Header
+        FITS header containing image information.
+
+    Returns
+    -------
+    list
+        Lower and upper Numpy array indices of the band.
+    """
+    y0_image_px = 1
+    y0_cen_px = header['CRPIX2']
+    y0_cen_rs = header['CRVAL2']
+    px_scale = header['CDELT2']
```

Comment:
Is this assuming `CDELT2` is in R_sun/pixel?

Reply:
Yes. Need adaptation if mosaic header has different units.